packages/core/parcel-bundler/test/assets/js-asset.js

```diff
@@ -0,0 +1,39 @@
+const assert = require('assert');
+const JSAsset = require('../../src/assets/JSAsset.js');
+
+describe('JSAsset', () => {
+  it('should not match sourceMappingURL when not at the end of the bundle', async () => {
+    // Code example taken from livescript.js (issue #2408 in parcel-bundler)
+    // This snippet lead to JSAsset.js being mislead and incorrectly trying to
+    // load (due to false-positive match) sourcemap before fix was introduced
+    const codeExample = `
+      if ((ref$ = options.map) === 'linked' || ref$ === 'debug') {
+        mapPath = path.basename(outputFilename) + ".map";
+        result.code += "\n//# sourceMappingURL=" + mapPath + "\n";
+      } else {
+        result.code += "\n//# sourceMappingURL=data:application/json;base64," + bufferFrom(result.map.toString()).toString('base64') + "\n";
+      }
+    `;
+    const a = new JSAsset(__filename, {rootDir: '/root/dir'});
+    Object.assign(a, {
+      type: 'type',
+      contents: codeExample
+    });
+
+    assert(!a.matchSourceMappingURL());
+  });
+
+  it('should match sourceMappingURL when correctly inserted at end of the bundle', async () => {
```

Comment:
Could you add a test for inline sourcemaps?

Reply:
oh, good catch. I've added another test case. I also wonder if there are other cases we should cover...